include/crocoddyl/multibody/contacts/contact-2d.hxx

```diff
@@ -0,0 +1,124 @@
+///////////////////////////////////////////////////////////////////////////////
+// BSD 3-Clause License
+//
+// Copyright (C) 2018-2020, LAAS-CNRS, University of Edinburgh
+// Copyright note valid unless otherwise stated in individual files.
+// All rights reserved.
+///////////////////////////////////////////////////////////////////////////////
+
+namespace crocoddyl {
+
+template <typename Scalar>
+ContactModel2DTpl<Scalar>::ContactModel2DTpl(boost::shared_ptr<StateMultibody> state, const FrameTranslation& xref,
+                                             const std::size_t& nu, const Vector2s& gains)
+    : Base(state, 2, nu), xref_(xref), gains_(gains) {}
+
+template <typename Scalar>
+ContactModel2DTpl<Scalar>::ContactModel2DTpl(boost::shared_ptr<StateMultibody> state, const FrameTranslation& xref,
+                                             const Vector2s& gains)
+    : Base(state, 2), xref_(xref), gains_(gains) {}
+
+template <typename Scalar>
+ContactModel2DTpl<Scalar>::~ContactModel2DTpl() {}
+
+template <typename Scalar>
+void ContactModel2DTpl<Scalar>::calc(const boost::shared_ptr<ContactDataAbstract>& data,
+                                     const Eigen::Ref<const VectorXs>&) {
+  Data* d = static_cast<Data*>(data.get());
+  pinocchio::updateFramePlacement(*state_->get_pinocchio().get(), *d->pinocchio, xref_.frame);
+  d->v = pinocchio::getFrameVelocity(*state_->get_pinocchio().get(), *d->pinocchio, xref_.frame);
+  d->vw = d->v.angular();
+  d->vv = d->v.linear();
+
+  pinocchio::getFrameJacobian(*state_->get_pinocchio().get(), *d->pinocchio, xref_.frame, pinocchio::LOCAL, d->fJf);
+  d->Jc.row(0) = d->fJf.template row(0);
+  d->Jc.row(1) = d->fJf.template row(2);
+
+  d->a = pinocchio::getFrameAcceleration(*state_->get_pinocchio().get(), *d->pinocchio, xref_.frame);
+  Vector3s a03D = d->a.linear() + d->vw.cross(d->vv);
```

Comment:
You are allocating memory here. 
You can simply implement by hand the cross() operation and do d->a0[0] = d->a.linear()[0] + cross_implementation and so on.

Reply:
This a stack allocation, no worry at this level.